lib/index.js

```diff
@@ -88,6 +89,20 @@ internals.GoodTracerStream = class GoodSourceTracer extends Stream.Transform {
 
 internals.goodTracerStreamFactory = (server, options) => new internals.GoodTracerStream(server, options);
 
+const deleteKey = ({ cache, id }) => {
+    debug('deleting id: %s', id);
+    cache.del(id);
+};
+
+internals.delayCacheDeleteFactory = (delay) => (request) => {
+    const requestId = get(request, 'info.id', 'NOOP');
+    const scopedCache = get(request, ['server', 'plugins', NAME, 'cache']);
+
+    // clean up cache
+    debug('delay delete of id %s for %d', requestId, delay);
+    setTimeout(deleteKey, delay, { cache: scopedCache, id: requestId });
```

Comment:
You could replace this line with `scopedCache.ttl(requestId, delay);`.

You don't get explicit confirmation that something has been cleared from memory, but you leverage the cache's ttl functionality that already exists.

docs: https://www.npmjs.com/package/node-cache#change-ttl-ttl

Reply:
Good callout, although the concern I have is that we set a default culling period of 5 minutes. So, with a TTL of `0` there is a maximum of 5 minutes that that the value will persist. :: pulls up source code ::

https://github.com/node-cache/node-cache/blob/master/_src/lib/node_cache.coffee#L343-L346

So, you are on to something @mrlannigan in that we could pass `-1` to get the `del` to occur, but ultimately we need to delay the delete in some manner to ensure that the `good` log stream has access to the cache data when logging out `response` events or other events that may happen JUST after the `response` (we see this in testing).

So, the question is do we set `scopedCache.ttl(requestId, delay);` as you are stating combined with a more aggressive cache clear check setting (currently 5 minutes) down to something like `500 ms`?

Reply:
I should add that the original logic behind the default TTL and cache check were to allow for a long running request to persist in the cache safely.

ooooooooo.... what if we set a default TTL of something VERY high and a cache check of much lower (10 min TTL with 500 ms cache check). Then we have the response hook set a TTL of 1 second.

Reply:
This would result in a more frequent scan of the cached values (max 5000 at defaults): https://github.com/node-cache/node-cache/blob/master/_src/lib/node_cache.coffee#L541-L549

Reply:
I did some testing and I cannot get a reliable delete via the `ttl` and `checkPeriod` settings. For now, I am going to leave the callback delete in place.

Reply:
I guess I don't fully understand the need to force an explicit deletion of the cached information.